### PR TITLE
Topology anti-affinity

### PR DESCRIPTION
**Pull request by rsevilla87**
Implement topology anti-affinity logic to place client and server pods. This will help to keep results more stable.

Signed-off-by: Raul Sevilla <rsevilla@redhat.com>

### CODE REVIEW

**Comment by rsevilla87** [comment]
@chaitanyaenr @mohit-sheth thoughts on this?

File: workloads/network-perf/common.sh [hunk 1]
```diff
@@ -41,12 +41,17 @@ export_defaults() {
...
-    export client=$(oc get nodes -l node-role.kubernetes.io/worker,node-role.kubernetes.io/workload!="" -o jsonpath='{range .items[*]}{ .metadata.labels.kubernetes\.io/hostname}{"\n"}{end}' | tail -n 1)
+  # Get AZs from worker nodes
+  zones=($(oc get nodes -l node-role.kubernetes.io/workload!=,node-role.kubernetes.io/worker -o go-template='{{ range .items }}{{ index .metadata.labels "topology.kubernetes.io/zone" }}{{ "\n" }}{{ end }}' | uniq -u))
+  if [[ ${#zones[@]} -gt 1 ]]; then
+    export server=$(oc get node -l node-role.kubernetes.io/workload!=,node-role.kubernetes.io/worker,topology.kubernetes.io/zone=${zones[0]} -o jsonpath='{range .items[*]}{ .metadata.labels.kubernetes\.io/hostname}{"\n"}{end}')
+    export client=$(oc get node -l node-role.kubernetes.io/workload!=,node-role.kubernetes.io/worker,topology.kubernetes.io/zone=${zones[1]} -o jsonpath='{range .items[*]}{ .metadata.labels.kubernetes\.io/hostname}{"\n"}{end}')
     export pin=true
+  else
+    log "At least 2 worker nodes placed in different topology zones are required"
```
Comment (jtaleric) [inline workloads/network-perf/common.sh:51]: Hm, is it required?

Comment (jtaleric) [reply workloads/network-perf/common.sh:51]: similar to the `pin=true`, I propose we allow single zone, but have another flat introduced that we pass as metadata, `multi_az=true`....

**Comment by Harshith-umesh** [comment]
Results for e2e-benchmarking CI Tests
Workload                | Test                           | Result | Runtime  |
------------------------|--------------------------------|--------|----------|

**Comment by Harshith-umesh** [comment]
!!! Couldn't read commit file !!!

**Review by jtaleric — APPROVED** [review]
LGTM

**Pull request merged**